components/script/body.rs

```diff
@@ -165,6 +181,9 @@ impl TransmitBodyConnectHandler {
         // In case of the data being in-memory, send everything in one chunk, by-passing SpiderMonkey.
         if let Some(bytes) = self.in_memory.clone() {
             let _ = bytes_sender.send(bytes);
+            // Mark this body as `done` so that we can correctly make it as Done in next tick
```

Comment:
nit: "so that we can stop reading in the next tick, matching the behavior of the promise-based flow."